backend/de.metas.ui.web.base/src/main/java/de/metas/ui/web/attachments/DocumentAttachmentsRestController.java

```diff
@@ -183,50 +184,55 @@ private boolean isAllowDeletingAttachments()
 		final IDocumentAttachmentEntry entry = getDocumentAttachments(windowIdStr, documentId)
 				.getEntry(entryId);
 
-		final AttachmentEntry.Type type = entry.getType();
-		if (type == AttachmentEntry.Type.Data)
+		final AttachmentEntryType type = entry.getType();
+		switch (type)
 		{
-			return extractResponseEntryFromData(entry);
-		}
-		else if (type == AttachmentEntry.Type.URL)
-		{
-			return extractResponseEntryFromURL(entry);
-		}
-		else
-		{
-			throw new AdempiereException("Invalid attachment entry")
-					.setParameter("reason", "invalid type")
-					.setParameter("type", type)
-					.setParameter("entry", entry);
+			case Data:
+				return DocumentAttachmentRestControllerHelper.extractResponseEntryFromData(entry);
+			case URL:
+				return DocumentAttachmentRestControllerHelper.extractResponseEntryFromURL(entry);
+			case LocalFileURL:
+				return DocumentAttachmentRestControllerHelper.extractResponseEntryFromLocalFileURL(entry);
+			default:
+				throw new AdempiereException("Invalid attachment entry")
+						.setParameter("reason", "invalid type")
+						.setParameter("type", type)
+						.setParameter("entry", entry);
 		}
 	}
 
-	private static ResponseEntity<byte[]> extractResponseEntryFromData(@NonNull final IDocumentAttachmentEntry entry)
+	private static ResponseEntity<StreamingResponseBody> extractResponseEntryFromData(@NonNull final IDocumentAttachmentEntry entry)
```

Comment:
do we need this method? don't we have it in helper?

Reply:
yep.. leftovers :)